### PR TITLE
Adding dynamic error page

### LINKED ISSUE #264: Error page  does not exist
**Is your feature request related to a problem? Please describe.**
Error page doesn't work. 
Go to e.g. https://devadventcalendar.pl/test404, you get

> Nie można znaleźć tej strony na devadventcalendar.plNie znaleziono strony internetowej pod adresem https://devadventcalendar.pl/test404
> HTTP ERROR 404

**Describe the solution you'd like**
E.g. https://github.com/DevAdventCalendar/DevAdventCalendar/issues/32

**Describe alternatives you've considered**
Redirect to main page.

**Additional context**

image

### PR DESCRIPTION
**Pull request by KrystianKolad**
## Description

Hi. I added handling client errors( for now 404, but handling other errors will be easy to add with this). I used UseStatusCodePagesWithRedirects and Added param to Error page so we can choose appropriate response based on status code. If status code is not implemented we will show default page (which is typical for status code 500). I am not too familiar with creating views in asp (I work mostly on APIs) so it will be nice to see what do you think of it ;)

## Motivation and context

resolves #264

## How has this been tested?

I wrote tests for new provider and run everything locally to check if html is rendered correctly.

## Screenshots (if appropriate)
image
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml [hunk 1]
```diff
@@ -14,8 +16,7 @@
     <div class="container">
         <div class="row">
             <div class="col text-center">
-                <h1 class="mb-30">Ups... Coś poszło nie tak...</h1>
-                <p>Nasze elfy już nad tym pracują ;-)</p>
+                @Model.Message
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml]: could you use h1 tag here?

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Startup.cs [hunk 2]
```diff
@@ -49,7 +49,7 @@ public static void Configure(IApplicationBuilder app, IHostingEnvironment env)
             }
             else
             {
-                app.UseExceptionHandler("/Home/Error");
+                app.UseStatusCodePagesWithRedirects("/Error?sc={0}");
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Startup.cs]: maybe consider renaming _sc_ to _statusCode_?

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml (hunk 1 above)
Comment (WTobor) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml]: and update tests, too :) You can use String.contains(), not compare whole html text

Comment (KrystianKolad) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml]: You mean move h1 tag from resources here?

Comment (WTobor) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml]: no - put the @model.Message value in h1 tag

Comment (KrystianKolad) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Home/Error.cshtml]: I added h1 tag, and used contains in tests

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Startup.cs (hunk 2 above)
Comment (KrystianKolad) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Startup.cs]: Renamed sc to statusCode

**Pull request merged**